rpc/kate-rpc/src/system.rs

```diff
@@ -0,0 +1,87 @@
+use frame_system_rpc_runtime_api::{system_events_api::fetch_events_v1, SystemEventsApi};
+use jsonrpsee::{
+	core::{async_trait, RpcResult},
+	proc_macros::rpc,
+	types::error::ErrorObject,
+};
+use sp_api::ProvideRuntimeApi;
+use sp_core::H256;
+use sp_runtime::traits::Block as BlockT;
+use std::{marker::PhantomData, sync::Arc};
+
+#[rpc(client, server)]
+pub trait Api {
+	#[method(name = "system_fetchEventsV1")]
+	async fn fetch_events_v1(
+		&self,
+		params: fetch_events_v1::Params,
+		at: H256,
+	) -> RpcResult<fetch_events_v1::ApiResult>;
+}
+
+pub struct Rpc<C, Block>
+where
+	C: ProvideRuntimeApi<Block> + Send + Sync + 'static,
+	C::Api: frame_system_rpc_runtime_api::SystemEventsApi<Block>,
+	Block: BlockT,
+	<Block as BlockT>::Hash: From<H256>,
+{
+	pub client: Arc<C>,
+	_phantom: PhantomData<Block>,
+}
+impl<C, Block> Rpc<C, Block>
+where
+	C: ProvideRuntimeApi<Block> + Send + Sync + 'static,
+	C::Api: frame_system_rpc_runtime_api::SystemEventsApi<Block>,
+	Block: BlockT,
+	<Block as BlockT>::Hash: From<H256>,
+{
+	pub fn new(client: Arc<C>) -> Self {
+		Self {
+			client,
+			_phantom: PhantomData,
+		}
+	}
+}
+
+/// Error type for this RPC API.
+pub enum Error {
+	/// Generic runtime error.
+	RuntimeApiError,
+}
+
+impl Error {
+	pub fn runtime_api_error<'a>(msg: String) -> ErrorObject<'a> {
```

Comment:
Since the function is returning an `ErrorObject`, I think it's better to name it likewise & make it an associated function
```suggestion
	pub fn into_error_object<'a>(self, msg: String) -> ErrorObject<'a> {
		ErrorObject::owned(self.into(), msg, None::<()>)
       }
```

Reply:
You are right, thank you!